### DISCUSSION TITLE
.NET DbContextFactory instead of DbContext

### DISCUSSION THREAD

**Discussion by petrkasnal** [jasontaylordev/CleanArchitecture#688] [General]
Hi, 

Im stuck on one problem. I would like use DbContextFactory. But it's problem here. Because DbContext interface - IApplicationDbContext is declared in project Application. Real DbContext is in Infrastructure project. So i cannot use DbContextFactory because i cannot use something like this IDbContextFactory<IApplicationDbContext>. Can somebody help me please? I'm really distressed of this. Thank you :)

**Comment by lwestfall** [comment]
I just outlined my implementation in another discussion. In short, put it in your Infrastructure project :)

https://github.com/jasontaylordev/CleanArchitecture/discussions/690#discussioncomment-3465426

Edit: Just realized you're talking about DbContextFactory, not DesignTimeDbContextFactory - but I think the answer is the same (you'll want this in Infrastructure, not Application).
